### DISCUSSION TITLE
How to send a request to another server?

### DISCUSSION THREAD

**Discussion by nth347** [PortSwigger/BChecks#100] [Q&A] [answered]
My scenario: While testing on an UAT environment `http://uat-server.com/uat/some/api`, BChecks found some issues. I want BChecks to automatically validate the issues on Production environment `https://server.com/app/some/api` by sending HTTP requests that requires:
- Changing URL Path from  to `/uat/some/api` to `/app/some/api`
- Changing the requested server from `http://uat-server.com` to `https://server.com`

Is it possible with BChecks?

**Comment by ayadim** [comment]
am not sure but i guess u can send another request after first check and before that change the host and path and test..

**Reply by nth347** [reply]
I attempted this method, but the second request is being sent to the same server as the first request.

**Comment by Michelle-PortSwigger** [comment]
Hi

Can I just double-check I'm understanding your requirements correctly?

You have written a BCheck that successfully finds and reports an issue in one environment.
As part of the same BCheck you then want a request to be sent to a completely separate domain.

How many sites does the current BCheck find issues on? Would the second request always be to the same site? Would you know you have permission to send the request to the second site at the time you are testing the first site?

**Reply by nth347** [reply]
Answer 1: Suppose BCheck has identified 1 issue on a specific site, let's say the UAT environment.

Answer 2: I need to send a second request to a completely different site, such as the Production environment.

Answer 3: I have the necessary permissions to do so.

When BCheck reports an issue, while we can manually use Burp Repeater to test it in the Production environment, automating this process would be more efficient for us.

Thank you.

**Reply by Michelle-PortSwigger** [reply]
BChecks are run against a scan item, so requests will be sent to the same host/application as the initial scan check rather than being expanded to send requests to hosts that are not part of the scan config.

Once the scan has run against your UAT environment you could potentially disable any BChecks that have not generated any results and then perform a BCheck only scan against your production environment which should reduce the time it takes to run the checks.

**Reply by nth347** [reply]
We all know that scanning against production environments is a sensitive action. I'm curious why BCheck allows changes to many aspects of the original request but doesn't permit altering the host or application being scanned.

**Reply by Michelle-PortSwigger** [reply, marked as answer]
BChecks are performing the same function as the in-built Burp Scanner checks, where requests are sent to the specific application that is in scope for a scan. They're not including additional functionality to automate further steps within a workflow that would send requests to applications that were not part of the original scan scope.

I hope this helps to explain things.
